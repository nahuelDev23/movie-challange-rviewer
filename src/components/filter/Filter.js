import React, { useContext } from 'react'
import { UiContext } from '../../context/ui/UiContext'

export const Filter = ({ category = 'movies' }) => {

    const { isOpenFilter } = useContext(UiContext)
    return (
        <>
            {
                isOpenFilter && (
                    <div className={`filter`}>
                        <form className='filter__form'>
                            <input className='filter__inputtext' type="text" placeholder='Name' />
                            <input className='filter__inputdate' type="date" />
                        </form>
                    </div>
                )
            }
        </>
    )
}
