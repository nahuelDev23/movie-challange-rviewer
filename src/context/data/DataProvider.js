import React, { useReducer } from 'react'
import api from '../../data'
import { DataContext } from './DataContext'
import { dataReducer } from './dataReducer'

const INITIAL_DATA_STATE = {
  series: [],
  movies: []
}

export const DataProvider = ({ children }) => {
  const [state, dispatch] = useReducer(dataReducer, INITIAL_DATA_STATE)

  const onLoadData = async (typeData) => {

    const data = await api.list()

    const filteredSeries = data[0].entries.filter(serie => serie.programType === typeData)

    const typeForDispatch = typeData === 'series' ? 'SET_SERIES' : 'SET_MOVIES'

    dispatch({
      type: typeForDispatch,
      payload: [...filteredSeries]

    })

  }


  return (
    <DataContext.Provider value={{
      ...state,

      //methods
      onLoadData

    }}>
      {children}
    </DataContext.Provider>
  )
}
