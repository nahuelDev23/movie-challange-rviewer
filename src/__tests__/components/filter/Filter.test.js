import { mount } from 'enzyme'
import { Filter } from '../../../components/filter/Filter';
import { DataContext } from '../../../context/data/DataContext';
import { UiContext } from '../../../context/ui/UiContext';

describe('Pruebas en <Filter/>', () => {

    test('Debe mostrar <Filter/> correctamente', () => {
        const wrapper = mount(
            <DataContext.Provider value={{
                onFilterDataByNameAndCategory: jest.fn()
            }}>
                <UiContext.Provider value={{
                    isOpenFilter: true
                }}>
                    <Filter category='movie' />
                </UiContext.Provider>
            </DataContext.Provider>

        );
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe captar el input text', () => {
        //!no verifica nada..
        const wrapper = mount(
            <DataContext.Provider value={{
                onFilterDataByNameAndCategory: jest.fn()
            }}>
                <UiContext.Provider value={{
                    isOpenFilter: true
                }}>
                    <Filter category='movie' />
                </UiContext.Provider>
            </DataContext.Provider>

        );

        const inputText = wrapper.find('input')
        inputText.simulate('change', { target: { value: 'Iron' } })


    })

  

})