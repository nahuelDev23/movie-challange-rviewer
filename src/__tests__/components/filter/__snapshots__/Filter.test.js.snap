// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Filter/> Debe mostrar <Filter/> correctamente 1`] = `
<div
  className="bg-primary"
>
  <div
    className="filter center-content"
  >
    <div
      className="filter__grouptext"
    >
      <div
        className="filter__icon"
      >
        Array [
          "",
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1q13jnw-MuiSvgIcon-root"
            data-testid="SearchOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>,
        ]
      </div>
      <input
        className="filter__inputtext"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
      />
    </div>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <button
          className="customDate"
          onClick={[Function]}
        >
          <div
            className="customDate__icon"
          >
            Array [
              "",
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarTodayIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                />
              </svg>,
            ]
          </div>
          <div
            className="customDate__text"
          >
            2022
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
